scalalib/src/ScalaModule.scala

```diff
@@ -128,17 +131,19 @@ trait ScalaModule extends JavaModule { outer =>
     resolveDeps(T.task{runIvyDeps() ++ scalaLibraryIvyDeps() ++ transitiveIvyDeps()})()
   }
 
-  override def compile: T[mill.scalalib.api.CompilationResult] = T.persistent{
+  override def compile: T[mill.scalalib.api.CompilationResult] = T.persistent {
+
     zincWorker.worker().compileMixed(
       upstreamCompileOutput(),
       allSourceFiles().map(_.path),
       compileClasspath().map(_.path),
       javacOptions(),
       scalaVersion(),
       scalaOrganization(),
-      scalacOptions(),
+      scalacOptions() ++ T.ctx.bsp.args,
```

Comment:
Using the contexts value directly means, we don't track it's value and potential changes in the cache. This may lead to missing re-compiles. Wrapping the args in a separate input task might fix this.

Reply:
Nope, you need to put it in a `def x = T.input{T.ctx.bsp.args}` and use that